src/storage/index/in_mem_hash_index.cpp

```diff
@@ -29,7 +29,9 @@ InMemHashIndex<T>::InMemHashIndex(OverflowFileHandle* overflowFileHandle)
       pSlots{std::make_unique<InMemDiskArrayBuilder<Slot<T>>>(dummy, 0, 0, true)},
       oSlots{std::make_unique<InMemDiskArrayBuilder<Slot<T>>>(dummy, 0, 1, true)},
       indexHeader{TypeUtils::getPhysicalTypeIDForType<T>()} {
-    allocatePSlots(1u << this->indexHeader.currentLevel);
+    // Match HashIndex in allocating at least one page of slots so that we don't split within the
+    // same page
+    allocateSlots(BufferPoolConstants::PAGE_4KB_SIZE / sizeof(Slot<T>));
```

Comment:
Should we assert that `sizeof(Slot<T>)` is power of 2 and less than `PAGE_4KB_SIZE`?

Reply:
We should assert that it's less than 4KB, but that's something the disk array should assert for any type stored in it.

But some slots are actually not powers of two, e.g. `Slot<uint8_t>` is actually only 212 bytes (limited to 20 entries by the fingerprint capacity), `Slot<uint16_t>` is 232, `Slot<uint32_t>` and `Slot<ku_string_t>` are 248, and then `Slot<uint64_t>` is 256.
I guess that means that this calculation doesn't work for exactly one page of slots as the disk array aligns elements to powers of two, meaning that all of them store 16 slots per page, but this reserves 20 slots for a `HashIndex<uint8_t>`.

Reply:
I've updated it to use the size rounded up to the nearest power of two from the disk array, and added the maximum size assertion to the disk array.